### PR TITLE
Feature samples selection

### PR DESCRIPTION
**Pull request by adityaekawade**

### CODE REVIEW

File: app/dataSelect.js [hunk 1]
```diff
@@ -94,7 +95,9 @@ DataSelect.prototype.addBuildListener = function() {
 				return;
 			}
 			genomeBuildHelper.setCurrentBuild(value);
+			//console.log("build val", value);
```
Comment (chmille4) [inline app/dataSelect.js]: Can remove this debug statement

**Comment by chmille4** [comment]
Lot of changes, so I didn't go through it with a fine tooth comb, but it looks good to me. I tested it as well and it looks good, but @tonydisera could you test it as well just to make sure. Then we should be able to merge

**Comment by chmille4** [comment]
Actually just found a bug. If you specify the .tbi file then two sampling events are fired. If you click the link below you can see the read sampled number go higher then lower then higher multiple times indicating two sampling commands have been executed.

http://localhost:3000/?species=Human&build=GRCh37&vcf=http%3A%2F%2Fs3.amazonaws.com%2Fvcf.files%2FExAC.r0.2.sites.vep.vcf.gz&tbi=http%3A%2F%2Fs3.amazonaws.com%2Fvcf.files%2FExAC.r0.2.sites.vep.vcf.gz.tbi

**Comment by tonydisera** [comment]
I did some testing this morning.  Overall, it is working well.  Here are a few minor issues:
1.  When the vcf is specified in the URL, vcf.iobio loads correctly.  However, the 'filter samples' button doesn't appear.
2.  When the .tbi file is in a different URL, the 'Loading...' message shows before I've had a chance to enter the tbi URL.
screen shot 2018-01-26 at 8 34 26 am

**Comment by adityaekawade** [comment]
@tonydisera :  Regarding (1), the filter sample button would appear only if the URL contains the samples. If it does not contain the samples, it would show just the load button. 
I am attaching the screenshots of both the cases. 

screen shot 2018-01-26 at 9 35 25 am
screen shot 2018-01-26 at 9 35 48 am

**Comment by tonydisera** [comment]
Aditya,
I am referring to the use case when the VCF URL is specified in the launch url.  When vcf.iobio is launched, the 'Filter samples' doesn't appear even though the vcf url is a multi-sample vcf:
http://localhost/vcf.iobio.io/?vcf=https%3A%2F%2Fs3.amazonaws.com%2Fiobio%2Fsamples%2Fvcf%2Fplatinum-exome.vcf.gz&tbi=&samples=NA12878&build=GRCh37

**Comment by adityaekawade** [comment]
Ok. Got it! I noticed the same thing now even when the page is refreshed. Thank you.

File: app/dataSelect.js (hunk 1 above)
Comment (adityaekawade) [reply app/dataSelect.js]: Hi, I have removed all the debug statements which were left in the latest commit.

**Comment by adityaekawade** [comment]
I have fixed these issues in the latest commit: 

1. Fixed the bug of two sampling commands fired when the tbi URL is specified. The read sampled number does not go higher-lower-higher now. 
2. The loading message appears only when both the URLs are entered. 
3. The filter sample button is now displayed when the vcf is specified in the url or when the page is refreshed. 

@tonydisera  @chmille4

**Comment by tonydisera** [comment]
Hi Aditya -

It looks like this fix may have broken something.  When I enter a multi-sample vcf (from the landing page), the sample select dropdown no longer appears.  
screen shot 2018-01-29 at 9 06 24 am


After clicking on load, the 'Filter samples' button does appear on the main page.



Launching vcf.iobio with the VCF URL now shows the 'Filter samples' button.
screen shot 2018-01-29 at 9 08 34 am

**Comment by adityaekawade** [comment]
Hi Tony, 
Can you please confirm again, it is working fine for me. 

screen shot 2018-01-29 at 9 26 31 am

**Comment by tonydisera** [comment]
Oops!  You are right.  It is fixed.  Looks good.  You can merge to gh-pages (and master) to deploy to production.

**Comment by tonydisera** [comment]
I just accepted this pull request and merged to gh-pages.  Aditya, can you please confirm that everything is working on production?  Thanks!

**Comment by adityaekawade** [comment]
Yes, I am checking that! Thank you.

**Comment by adityaekawade** [comment]
It's working fine on production!

**Pull request merged**